services/execution/service.go

```diff
@@ -477,6 +478,9 @@ func (s *Service) writeContent(ctx context.Context, mediaType, ref string, r io.
 }
 
 func (s *Service) getTask(ctx context.Context, id string) (plugin.Task, error) {
+	if _, err := namespaces.NamespaceRequired(ctx); err != nil {
+		return nil, grpc.Errorf(codes.InvalidArgument, "task %v not found: %v", id, err.Error())
```

Comment:
We should just get the container metadata and then that will have the runtime on it to grab the task.

container := store.GetContainer(ctx, id)
rt, err := getRuntime(container.Runtime)
task, err := rt.Task(id)